projects/ad3552r_fmcz/srcs/main.c

```diff
@@ -0,0 +1,211 @@
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <inttypes.h>
+#include "app_config.h"
+#include "parameters.h"
+#include "no-os/error.h"
+#include "no-os/print_log.h"
+#include "no-os/spi.h"
+#include "no-os/gpio.h"
+#include "no-os/util.h"
+#include "no-os/delay.h"
+#include "spi_extra.h"
+#include "gpio_extra.h"
+#include "ad3552r.h"
+
+#include <xparameters.h>
+#include <xil_cache.h>
+#include <xil_cache.h>
+
+/* Default structures */
+static struct xil_gpio_init_param xil_gpio_param = {
+	.device_id = GPIO_DEVICE_ID,
+	.type = GPIO_PS
+};
+
+static struct gpio_init_param default_gpio_param = {
+	.platform_ops = &xil_gpio_ops,
+	.extra = &xil_gpio_param
+};
+
+int32_t init_gpios_to_defaults()
+{
+	static uint8_t gpios_initial_value[][2] = {
```

Comment:
Why have it static?

Reply:
If first used it as global, then I forgot to remove it